### ISSUE TITLE
Namespaces in arguments of a method breaks cref resolving in C#

### ISSUE THREAD

**Issue by klara-zielinska** [doxygen/doxygen#9669]
I have a method `int Show(System.Windows.Window parent)` and  the `<see cref="Show(System.Windows.Window)"/>` comment reference in the same C# class. Doxygen doesn't resolve the reference. 

If I change `System.Windows.Window` to `Window` (which is a different class) in both places, the reference is being resolved.

**Comment by albert-github** [comment]
- Can you please attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file) that allows us to reproduce the problem? Please don't add external links as they might not be persistent.
- Please also specify the **full** doxygen version used (`doxygen -v`).
  From your previous issue (#9666) I assume that it is 1.9.5

**Comment by klara-zielinska** [comment]
[DoxyIssue_NamespacedArgInCref.zip](https://github.com/doxygen/doxygen/files/9974132/DoxyIssue_NamespacedArgInCref.zip)
Version: 1.9.5 (2f6875a5ca481a69a6f32650c77a667f87d25e88)
Doxyfile in the project directory.

**Comment by albert-github** [comment]
I don't catch the remark 
> If I change System.Windows.Window to Window (which is a different class) in both places, the reference is being resolved.

can you give an example of this?

I looked at the source code and especially at
```
        /// <summary>
        /// <see cref="Show(System.Windows.Window)"/>
        /// </summary>
```
When I would replace the `<see cref="Show(System.Windows.Window)"/>` with `\ref <see cref="Show(System.Windows.Window)"` I would get a link to the second (right) function.
When I replace `<see cref="Show(System.Windows.Window)"/>` with `<see cref="Show(System.Windows.Window)">Show(System.Windows.Window)</see>` I will get also a link, though this link is not pointing to the right place (it just points to the page and not the the second function).

I also see warnings like:
```
Window.cs:13: warning: unable to resolve link to 'Show(System.Windows.Window)' for \link command
```
which brings me to the idea if incorrect class / namespace reference for C#, when changing the in the definition the dot (`.`) into a double colon (`::`) all looks like to work.

See also the small example:  [example.tar.gz](https://github.com/doxygen/doxygen/files/9979688/example.tar.gz)

**Comment by klara-zielinska** [comment]
> I don't catch the remark
> 
> > If I change System.Windows.Window to Window (which is a different class) in both places, the reference is being resolved.
> 
> can you give an example of this?

Replace 

`<see cref="Show(System.Windows.Window)"/>` with `<see cref="Show(Window)"/>`

and the method signature

`Show(System.Windows.Window parent)` with `Show(Window parent)`.

---

> which brings me to the idea if incorrect class / namespace reference for C#, when changing the in the definition the dot (`.`) into a double colon (`::`) all looks like to work.

Yes, changing `.` to `::` in both `cref` and the method signature works. However `::` does not compile in C#.

I guess, it is a shortcoming, because in another scenario the `.` symbol works fine as the namespace accessor. E.g.
```
namespace X
{
    public class Window
    {
        /// <summary>
        /// <see cref="X.Window.Show(Window)"/>
        /// </summary>
        public void Show()
        { }

        /// <summary>
        /// </summary>
        public void Show(Window parent)
        { }
    }
}
```

**Comment by albert-github** [comment]
Thanks for the extra examples in https://github.com/doxygen/doxygen/issues/9669#issuecomment-1310112484, seen these both examples it looks like the problem is in respect to the function parameter type(s).

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #9671

**Comment by klara-zielinska** [comment]
When can be the merge expected?

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by klara-zielinska** [comment]
Thank you. I am impressed with such a quick response.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.6.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
